sds/src/secondary_validation/jwt_claims_validator.rs

```diff
@@ -35,17 +36,34 @@ impl JwtClaimsValidator {
                 patterns.insert(claim_name.clone(), Regex::new(pattern).unwrap());
             }
         }
+
         Self {
-            required_claims: config.required_claims.into_iter().collect(),
+            header_required_claims: config
+                .required_claims
+                .clone()
+                .into_iter()
+                .filter(|(claim_name, _)| claim_name.starts_with("header."))
+                .collect(),
+            payload_required_claims: config
+                .required_claims
+                .into_iter()
+                .filter(|(claim_name, _)| !claim_name.starts_with("header."))
+                .collect(),
```

Comment:
This will contribute to create a lot of magic around the format of headers. The configuration is not already used so I think it's better to update the config format and keep the logic here simpler than adding this blurry logic which will required additional context to use the validator and make the learning curve to use this validator steeper.